### PR TITLE
Fix post upgrade bug

### PR DESCRIPTION
**Pull request by timizuoebideri1**
# Closes: #4604 
# What's Changed

# TODO

- [ ] Explanation of Change(s)
- [ ] Added change log fragment(s) (for more information see [the documentation](https://docs.nautobot.com/projects/core/en/stable/development/#creating-changelog-fragments))
- [ ] Attached Screenshots, Payload Example
- [ ] Unit, Integration Tests
- [ ] Documentation Updates (when adding/changing features)
- [ ] Example Plugin Updates (when adding/changing features)
- [ ] Outline Remaining Work, Constraints from Design

### CODE REVIEW

**Review by glennmatthews — COMMENTED** [review]
Did you audit the other removed model migrations (https://github.com/nautobot/nautobot/issues/4604#issuecomment-1747649995) to see if similar fixes may be needed there?

File: nautobot/dcim/migrations/0029_device_and_rack_roles_data_migrations.py [hunk 1]
```diff
@@ -32,6 +42,15 @@ def reverse_role_data_migrate(apps, schema):
             to_role_field_name="legacy_role",
             to_role_model=to_role_model,
         )
+    update_object_change_ct_for_replaced_models(
+        apps=apps,
+        new_app_model={"app_name": "dcim", "model": "role"},
```
Comment (timizuoebideri1) [inline nautobot/dcim/migrations/0029_device_and_rack_roles_data_migrations.py:47]:
```suggestion
        new_app_model={"app_name": "extras", "model": "role"},
```

File: nautobot/core/utils/migrations.py [hunk 2]
```diff
@@ -33,3 +33,30 @@ def check_for_duplicates_with_natural_key_fields_in_migration(model_class, natur
             file=sys.stderr,
         )
         raise RuntimeError("Duplicate records must be manually resolved before migrating.")
+
+
+def update_object_change_ct_for_replaced_models(apps, new_app_model, replaced_apps_models, reverse_migration=False):
```
Comment (glennmatthews) [inline nautobot/core/utils/migrations.py:38]: Unit test looks reasonable - can we also add representative data to the `development/datasets/` DB dumps used in our CI migration tests as a regression test? From a quick look it looks like those dumps don't currently include any ObjectChange records at all.

Comment (glennmatthews) [reply nautobot/core/utils/migrations.py:38]: I've opened an issue against the private repo that the datasets were generated from, asking that it add such data.

**Review by glennmatthews — APPROVED** [review]

**Pull request merged**